src/frontend/src/lib/types/dapp.ts

```diff
@@ -0,0 +1,34 @@
+import { z } from 'zod';
+
+// see https://github.com/dfinity/portal/tree/95c67a5cfe201e4e5cb79f3cf5d18fe16498cd8c?tab=readme-ov-file#object-schema
+const dAppSchema = z.object({
+	id: z.string(),
+	name: z.string(),
+	// TODO replicate logic from https://github.com/dfinity/portal/blob/34a0328ed4792f5a7f3943be73f13f5abaefb4b8/plugins/validate-showcase.js#L179
+	oneLiner: z.string(),
+	// TODO validate that this URL starts with HTTPs
+	website: z.string().url(),
+
+	tags: z.array(z.string()),
+	description: z.string(),
+	stats: z.string(),
+	logo: z.string(),
+
+	usesInternetIdentity: z.boolean(),
+	authOrigins: z.array(z.string()).optional(),
+
+	github: z.string().url().optional(),
+	youtube: z.string().url().optional(),
+	twitter: z.string().url().optional(),
+
+	screenshots: z.array(z.string()).optional(),
+
+	video: z.string().optional(),
+	videoContentType: z.enum(['video/webm', 'video/mp4']).optional(),
+
+	submittableId: z.string().optional(),
+
+	featured: z.boolean().optional()
+});
+
+export type DApp = z.infer<typeof dAppSchema>;
```

Comment:
@peterpeterparker @loki344 what do you think of renaming this `DAppInfo` or something like that?

Reply:
In Internet Identity they actually use `DappDescription` for the JSON data and a wrapper `KnownDapp` for those that have been asserted given that they fetch the list from the portal. 

So maybe `DappDescription` as well?

Ultimately we can use the same zod parser in II and Oisy and Portal if it's the same structure and if the foundation wants to avoid another mess.

II source code: https://github.com/dfinity/internet-identity/blob/a537c50f9707034ff4d2cfc41d77ff8caa0a845e/src/frontend/src/flows/dappsExplorer/dapps.ts#L15C13-L15C28

Reply:
IMHO adding things like "Info" or "Data" to variable names rarely adds any value. I will use DappDescription, even though I think the interface is more than just a description (the field description inside of it implies that the description is part of the interface).